### PR TITLE
[lcm] Fix/lcm tools

### LINKED ISSUE #24408: [lcm] Fix tooling support in Linux/macOS
****

**Is your feature request related to a problem? Please describe.**
Right now, the `lcm` portfile searches for `.exe` files to install as tools, which don't exist in non-Windows systems. Downstream users of the lib, when including the LCMTargets/config.cmake files will get an error that these targets are called out but don't exit

**Proposed solution**
Modify the port file to explicitly name the installed tools, copy these to the `tools/lcm` folder

**Describe alternatives you've considered**
Could alternatively dynamically search for the targets in the binary directory

**Additional context**
None

### PR DESCRIPTION
**Pull request by venabled**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #24408 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  All triplets should be supported, have not updated the ci baseline

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  <Yes>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by Adela0814** [comment]
Are you trying to fix the build error of Linux？You can click this link to get the error log: https://dev.azure.com/vcpkg/public/_build/results?buildId=71346&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by venabled** [comment]
I think it just timed out?

File: triplets/x64-linux.cmake [hunk 1]
```diff
@@ -1,6 +1,6 @@
 set(VCPKG_TARGET_ARCHITECTURE x64)
 set(VCPKG_CRT_LINKAGE dynamic)
-set(VCPKG_LIBRARY_LINKAGE static)
+set(VCPKG_LIBRARY_LINKAGE dynamic)
```
Comment (Adela0814) [inline triplets/x64-linux.cmake:3]: This cannot be changed because it will affect all ports. Please restore it, thanks.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: triplets/x64-linux.cmake (hunk 1 above)
Comment (venabled) [reply triplets/x64-linux.cmake]: Sorry, forgot that was in there. Fixed

**Comment by Adela0814** [comment]
Error message:
```
CMake Error at scripts/cmake/vcpkg_copy_tools.cmake:70 (message):
  Couldn't find tool "lcm-example":

      "/mnt/vcpkg-ci/packages/lcm_x64-linux/bin/lcm-example" does not exist
```

**Comment by PhoebeHui** [comment]
@venabled, could you please take a look at the failures? it looks no tools that generated on linux.

**Comment by Adela0814** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**